controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -129,6 +142,37 @@ func (a *Adapter) EnsureSnapshotExists() (controller.OperationResult, error) {
 		return controller.RequeueWithError(err)
 	}
 
+	// Remove before merge
+	a.logger.Info("Reached end of EnsureSnapshotExists()")
+	return a.removeFinalizerAndContinueProcessing()
+}
+
+func (a *Adapter) EnsurePipelineIsFinalized() (controller.OperationResult, error) {
+	// Remove before merge
+	a.logger.Info(fmt.Sprintf("Adding finalizer '%s' to build pipeline run '%s'", finalizer, a.pipelineRun.Name))
+	if !h.HasPipelineRunFinished(a.pipelineRun) || controllerutil.ContainsFinalizer(a.pipelineRun, finalizer) {
```

Comment:
```suggestion
	if h.HasPipelineRunFinished(a.pipelineRun) || controllerutil.ContainsFinalizer(a.pipelineRun, finalizer) {
```
To allow the finalizer to be added on the build PLR right after it gets created